sdk/keyvault/keyvault-keys/src/index.ts

```diff
@@ -1266,11 +1268,16 @@ export class KeyClient {
    */
   private setParentSpan(span: Span, options: RequestOptionsBase = {}): RequestOptionsBase {
     if (span.isRecording()) {
+      const spanOptions = options.spanOptions || {};
       return {
         ...options,
         spanOptions: {
-          ...options.spanOptions,
-          parent: span
+          ...spanOptions,
```

Comment:
Probably not for this PR but perhaps we need to start thinking about common utility functions we export from core-tracing. Every time we expand this it gets harder and harder to track. It could also help us enforce some constraints that are important to us, like always specifying the az.namespace, etc...

Is that an anti-goal of the tracing package? Would it be better as some sort of "internal only" package?

Reply:
I think it's a good goal. I filed #6844 to track it